### PR TITLE
ISPN-13459 rolling upgrade doc additions

### PR DESCRIPTION
**Pull request by oraNod**
https://issues.redhat.com/browse/ISPN-13459 

Needs backport to 13.0.x

### CODE REVIEW

File: documentation/src/main/asciidoc/stories/assembly_server_rolling_upgrades.adoc [hunk 1]
```diff
@@ -1,13 +1,12 @@
...
-:context: upgrade
-= Performing Rolling Upgrades for {brandname} Servers
+[id='rolling-upgrades']
+:context: rolling-upgrades
+= Performing rolling upgrades for {brandname} Server clusters
 Perform rolling upgrades of your {brandname} clusters to change between
-versions without downtime or data loss. Rolling upgrades migrate both your
-{brandname} servers and your data to the target version over Hot Rod.
+versions without downtime or data loss.
+Rolling upgrades migrate your {brandname} Server instances as well as your data to the target version over Hot Rod.
```
Comment (gustavonalle) [inline documentation/src/main/asciidoc/stories/assembly_server_rolling_upgrades.adoc:6]: What does "migrate your {brandname} Server instances" mean? Rolling Upgrades on bare metal only migrates the data, it does not deal with configuration, starting/stop/creating servers

File: documentation/src/main/asciidoc/topics/proc_setting_up_target_clusters.adoc [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+[source,bash,options="nowrap",subs=attributes+]
+----
+[//containers/default]> migrate cluster connect -c myCache -f remote-store.json
+----
++
+* REST API: Use the `?action=connect-source` parameter with a POST request that includes the remote store configuration in the payload.
++
+[source,options="nowrap",subs=attributes+]
+----
+POST /v2/caches/myCache?action=connect-source
```
Comment (gustavonalle) [inline documentation/src/main/asciidoc/topics/proc_setting_up_target_clusters.adoc:42]: This method doesn't exist, the source connection handling methods are 
https://github.com/infinispan/infinispan/blob/7e833c66558279e9adb750487fe7c254f266b48c/server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java#L150-L153

**Review by gustavonalle — CHANGES_REQUESTED** [review]

File: documentation/src/main/asciidoc/stories/assembly_server_rolling_upgrades.adoc (hunk 1 above)
Comment (oraNod) [reply documentation/src/main/asciidoc/stories/assembly_server_rolling_upgrades.adoc:6]: nice catch, thanks. I've reworded this.

File: documentation/src/main/asciidoc/topics/proc_setting_up_target_clusters.adoc (hunk 2 above)
Comment (oraNod) [reply documentation/src/main/asciidoc/topics/proc_setting_up_target_clusters.adoc:42]: nuts. I should have double checked this one, sorry. shame on me for only trying the CLI. `?action=connect-source` was already in the docs so I just took it that was OK. I've fixed it now.

**Review by gustavonalle — APPROVED** [review]

**Pull request merged**